lua/r/debug.lua

```diff
@@ -0,0 +1,147 @@
+------------------------------------------------------------------------------
+-- Support for debugging R code
+------------------------------------------------------------------------------
+local config = require("r.config").get_config()
+
+local r_bufnr
+local M = {}
+
+-- Handle ambiwidth option and define signs
+if vim.o.ambiwidth == "double" then
+    vim.fn.sign_define(
+        "dbgline",
+        { text = "==>", texthl = "SignColumn", linehl = "QuickFixLine" }
+    )
+else
+    vim.fn.sign_define(
+        "dbgline",
+        { text = "▬▶", texthl = "SignColumn", linehl = "QuickFixLine" }
+    )
+end
+
+local s = {
+    func_offset = -2, -- Did not seek yet
+    rdebugging = 0
+}
+
+M.stop_r_debugging = function()
+    vim.fn.sign_unplace("rnvim_dbgline", { id = 1 })
+    s.func_offset = -2
+    s.rdebugging = 0
+end
+
+M.find_debug_func = function(srcref)
+    local rlines
+    local can_debug = type(config.external_term) == "boolean" and not config.external_term
+    if not can_debug then return end
+
+    s.func_offset = -1 -- Not found
+    local sbopt = vim.o.switchbuf
+    vim.o.switchbuf = "useopen,usetab"
+    local curtab = vim.fn.tabpagenr()
+    local isnormal = vim.fn.mode() == "n"
+    local curwin = vim.fn.winnr()
+    r_bufnr = vim.api.nvim_get_current_buf()
+    vim.cmd("sb " .. r_bufnr)
+    vim.fn.sleep(30) -- Time to fill the buffer lines
+    rlines = vim.fn.getline(1, "$")
```

Comment:
In this PR, the debug code only finds `rlines` (The R source lines I believe) if there is no `external_term` in the config. 

There was vimscript code found `rlines` when debugging and I could look to convert it. That code is here. 

https://github.com/R-nvim/R.nvim/blob/5b551860efe6a07b2e565aa24e124612e18c0a5e/R/start_r.vim#L97-L102

I searched the commit history last week but couldn't determine what the proper way to do this in the lua plugin. As such I removed it. I mentioned it in the initial comment that I removed some code paths that looked like unsupported tmux.

Be happy to be advised on how to fix.

Reply:
At this point, it's better to no longer look at `Nvim-R` code. What has to be done is relatively simple:

1. When beginning the debugging, load the file where the code is or jump to it if it's already open (this is already done in the specific case that I fixed) and store the buffer number (not done yet).
2. From the second jump onward, move the cursor without switching to its buffer (this will be different from `Nvim-R`).